### PR TITLE
Qa 2.23.0

### PR DESCRIPTION
**Pull request by lunkan93**
### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [ ] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [x] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Critical security fixes should be marked with `kind/security`
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### What does this PR do / why do we need this PR?

...

- Fixes #

#### Checklist

- [ ] Proper commit message prefix on all commits
  
- Change checks:
  - [ ] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco

### CODE REVIEW

File: rook/migration/v1.11/README.md [hunk 1]
```diff
@@ -75,6 +75,14 @@
 
     These will now be provided from here so they must be removed else they will be in conflict.
 
+    ```bash
+    yq4 -i '.rookCeph.monitoring.enabled = false' "$CK8S_CONFIG_PATH/common-config.yaml"
+    yq4 -i '.rookCeph.gatekeeperPsp.enabled = false' "$CK8S_CONFIG_PATH/common-config.yaml"
+    yq4 -i '.networkPolicies.rookCeph.enabled = false' "$CK8S_CONFIG_PATH/common-config.yaml"
```
Comment (OlleLarsson) [inline rook/migration/v1.11/README.md:81]: Should we create an issue to remove these from apps?

**Review by OlleLarsson — APPROVED** [review]

File: rook/migration/v1.11/README.md (hunk 1 above)
Comment (lunkan93) [reply rook/migration/v1.11/README.md:81]: create an issue for this
https://github.com/elastisys/compliantkubernetes-apps/issues/1824

File: config/common/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml [hunk 2]
```diff
@@ -54,6 +54,9 @@ calico_ipip_mode: 'Always'
 calico_vxlan_mode: 'Never'
 calico_network_backend: 'bird'
 
+coredns_additional_error_config: |
```
Comment (Xartos) [inline config/common/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml:57]: Maybe add a comment here saying this isn't in effect yet, just to make it clear
```suggestion
# This variable will not be present in kubespray in 2.23.0, but will be in releases afterwards
coredns_additional_error_config: |
```

Comment (Xartos) [reply config/common/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml:57]: Never mind this

**Review by Xartos — APPROVED** [review]

File: migration/v2.23/prepare/30-coredns-config.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -e
+
+additional_error_config='consolidate 5m ".* i/o timeout$" warning'
+
+tempfile=$(mktemp)
+trap 'rm -f "$tempfile"' EXIT
```
Comment (davidumea) [inline migration/v2.23/prepare/30-coredns-config.sh:8]:
```suggestion
trap 'rm -f "${tempfile}"' EXIT
```

File: migration/v2.23/prepare/30-coredns-config.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+printf "coredns_additional_error_config: |\n  %s\n" "$additional_error_config" > "$tempfile"
```
Comment (davidumea) [inline migration/v2.23/prepare/30-coredns-config.sh:10]:
```suggestion
printf "coredns_additional_error_config: |\n  %s\n" "${additional_error_config}" > "${tempfile}"
```

File: migration/v2.23/prepare/30-coredns-config.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+yq4 eval-all 'select(fileIndex == 0) * select(fileIndex == 1)' "$CK8S_CONFIG_PATH"/wc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml "$tempfile" -i
+yq4 eval-all 'select(fileIndex == 0) * select(fileIndex == 1)' "$CK8S_CONFIG_PATH"/sc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml "$tempfile" -i
```
Comment (davidumea) [inline migration/v2.23/prepare/30-coredns-config.sh:13]:
```suggestion
yq4 eval-all 'select(fileIndex == 0) * select(fileIndex == 1)' "${CK8S_CONFIG_PATH}"/wc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml "${tempfile}" -i
yq4 eval-all 'select(fileIndex == 0) * select(fileIndex == 1)' "${CK8S_CONFIG_PATH}"/sc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml "${tempfile}" -i
```

File: migration/v2.23/prepare/30-coredns-config.sh [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+rm -f "$tempfile"
```
Comment (davidumea) [inline migration/v2.23/prepare/30-coredns-config.sh:15]:
```suggestion
rm -f "${tempfile}"
```

File: migration/v2.23/upgrade-cluster.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+1. Switch to the correct remote: `git submodule sync`
+
+1. Update the kubespray submodule: `git submodule update --init --recursive`
+
+1. Set `ck8sKubesprayVersion` to `any` in `sc-config/group_vars/all/ck8s-kubespray-general.yaml` and `wc-config/group_vars/all/ck8s-kubespray-general.yaml`
+
+    ```bash
+    yq4 -i '.ck8sKubesprayVersion = "any"' ${CK8S_CONFIG_PATH}/sc-config/group_vars/all/ck8s-kubespray-general.yaml
+    yq4 -i '.ck8sKubesprayVersion = "any"' ${CK8S_CONFIG_PATH}/wc-config/group_vars/all/ck8s-kubespray-general.yaml
+    ```
```
Comment (aarnq) [inline migration/v2.23/upgrade-cluster.md:32]: Is there a bug or something in the upgrade prepare scripts, cause shouldn't this be properly managed by it?

Comment (Pavan-Gunda) [reply migration/v2.23/upgrade-cluster.md:32]: Yes, 

There seems to be a bug in the upgrade prepare scripts, they are not working until we set the version to any.

Comment (aarnq) [reply migration/v2.23/upgrade-cluster.md:32]: OK, we should definitely fix that for the next release.

**Review by aarnq — APPROVED** [review]

**Pull request merged**